plugins/tiddlywiki/multiwikiserver/modules/store/sql-tiddler-database.js

```diff
@@ -374,6 +491,68 @@ SqlTiddlerDatabase.prototype.getRecipeTiddler = function(title,recipe_name) {
 	};
 };
 
+/*
+Checks if a user has permission to access a recipe
+*/
+SqlTiddlerDatabase.prototype.hasRecipePermission = function(userId, recipeName, permissionName) {
+	return this.checkACLPermission(userId, 'recipe', recipeName, permissionName)
+};
+
+/*
+Checks if a user has permission to access a bag
+*/
+SqlTiddlerDatabase.prototype.hasBagPermission = function(userId, bagName, permissionName) {
+	return this.checkACLPermission(userId, 'bag', bagName, permissionName)
+};
+
+SqlTiddlerDatabase.prototype.checkACLPermission = function(userId, entityType, entityName, permissionName) {
+	const entityTypeToTableMap = {
+		bag: {
+			table: 'bags',
+			column: 'bag_name'
+		},
+		recipe: {
+			table: 'recipes',
+			column: 'recipe_name'
+		}
+	};
+
+	const entityInfo = entityTypeToTableMap[entityType];
+	if (!entityInfo) {
+		throw new Error('Invalid entity type: ' + entityType);
```

Comment:
I appreciate its a nuisance, but there are still a lot of string constants using single quotes instead of the preferred double quotes.